x-pack/platform/plugins/shared/encrypted_saved_objects/integration_tests/ci_checks/check_registered_types.test.ts

```diff
@@ -0,0 +1,308 @@
+      return map;
+    }, {} as Record<string, string>);
+
+    expect(hashMap).toMatchInlineSnapshot(`
+      Object {
+        "action": "696d997e420024a8cf973da94d905c8756e1177c",
+        "action_task_params": "cd91a48515202852ebf1fed0d999cd96f6b2823e",
+        "ad_hoc_run_params": "690b8991f48c73a04e6a8cf41fd4967a42f8e552",
+        "alert": "581f322afd6784aebdb80f872b825a928533d364",
+        "api_key_pending_invalidation": "cef0693ec88475a0e1f43614cfa6ca43c24d0338",
+        "connector_token": "e25821ecec3061806a6a9d4953273c18a749cc0f",
+        "entity-discovery-api-key": "094f1eae0e069e5f8bf2523db1a14072a8e29271",
+        "fleet-fleet-server-host": "795c0e79438a260bd860419454bcc432476d4396",
+        "fleet-message-signing-keys": "0c6da6a680807e568540b2aa263ae52331ba66db",
+        "fleet-uninstall-tokens": "216be68d8426052f9e7529e2e0569b7950676537",
+        "ingest-download-sources": "e6b6c76a67a1882c861177ee9e8ff2c607b7eeea",
+        "ingest-outputs": "f92200366d6b9f142a81f094154e17987910c535",
+        "privmon-api-key": "c06b1614786ce7271087378b47d465c956ab1537",
+        "synthetics-monitor": "fdebfa2449d2b934972d1743dc78c34ae9ebc9c1",
+        "synthetics-monitor-multi-space": "c8c9dab447ba8a7383041f55ba80757365d114c5",
+        "synthetics-param": "9776c9b571d35f0d0397e8915e035ea1dc026db7",
+        "task": "689edead32ea09558ceb54f64fd9aa4d324d94d0",
+        "uptime-synthetics-api-key": "599319bedbfa287e8761e1ba49d536417a33fa13",
+      }
+    `);
+    expect(Object.keys(hashMap).length).toEqual(ESO_TYPES_COUNT);
+  });
+
+  // This test is meant to more granularly detect what changes when occured to an ESO's migration-related properties.
+  // Though there is a similar test in core, this is relevent to ESOs and the Kibana Security team.
+  // Just update the snapshot by running this test file via jest_integration with `-u` and push the update.
+  // The intent is to trigger a code review from the Kibana Security team to review the ESO changes.
+  // The number of types in the hashMap should never be reduced, it should only increase.
+  it('detecting modelVersion and schema changes in registered types', () => {
+    const esoTypes = esoService.getRegisteredTypes();
+    const soTypestoCheck = typeRegistry
+      .getAllTypes()
+      .filter((soType) => esoTypes.includes(soType.name));
+
+    const hashMap: string[] = sortBy(soTypestoCheck, 'name').flatMap((type, index) => {
+      const typeHashes = getTypeHashes(type);
+
+      if (index < soTypestoCheck.length - 1) {
+        const length = typeHashes[typeHashes.length - 1].length;
+        if (length) {
+          typeHashes.push('='.repeat(length));
+        }
+      }
+      return typeHashes;
+    });
+
+    expect(hashMap).toMatchInlineSnapshot(`
+      Array [
+        "action|global: 04984aae6011426601f8a2a06278e30080f6da3a",
+        "action|mappings: c4a658c865d4c30b51ae9b49e1dec06d012bc213",
+        "action|schemas: da39a3ee5e6b4b0d3255bfef95601890afd80709",
```

Comment:
I don't think we need to include global, mappings, or schemas here - we could filter them out of the array before comparing.